app/models/page.coffee

```diff
@@ -10,16 +10,28 @@ PageModel = Backbone.Model.extend(
         selected: false
         dragging: false
         ordinal: -1
-
+        viewed: false
+        loading: false
+        loaded: false
 )
 
 # Page Collection
 PagesCollection = Backbone.Collection.extend(
+    _toBeLoadedQueue: []
+    _loadingSlotAvailable: ->
+        return not @maxConcurrentLoadingPages or @where({loading: true}).length < @maxConcurrentLoadingPages
```

Comment:
You can remove `not @maxConcurrentLoadingPages or` if you use `Infinity` (see my comment above)